### PR TITLE
Update error group callback api implementation

### PR DESCRIPTION
**Pull request by nrcventura**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description

Updates the code referenced in #1400 
- Renames the API for ErrorFingerprintingCallback to SetErrorGroupCallback.
- Updates the API SetErrorGroupCallback implementation to behave similarly to the SetApplicationName API method
- Adds a documentation summary to the SetErrorGroupCallback method

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)
- ~~[ ] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated~~ Will be updated in a separate PR 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

File: src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/InternalApi.cs [hunk 1]
```diff
@@ -317,13 +317,16 @@ public static void SetApplicationName(string applicationName, string? applicatio
...
-        /// DOCS GO HERE
+        /// <summary> Sets the method that will be invoked to define the error group that an exception
+        /// should belong to.
+        ///
+        /// The callback takes an Exception and returns the name of the error group to use. Return values
+        /// that are null, empty, or whitespace will not associate the Exception to an error group.
         /// </summary>
-        /// <param name="callback"></param>
+        /// <param name="callback">The callback to invoke to define the error group that an Exception belongs to.</param>
         public static void ErrorFingerprintingCallback(Func<Exception, string> callback)
```
Comment (chynesNR) [inline src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/InternalApi.cs:327]: We should probably rename this to match the others?

**Comment by chynesNR** [comment]
Looks good!

File: src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/InternalApi.cs (hunk 1 above)
Comment (nrcventura) [reply src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/InternalApi.cs:327]: Yes, good catch. I'll do a search to see if there are others that need renamed.

Comment (nrcventura) [reply src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/InternalApi.cs]: After c29f60f the remaining references to the fingerprint callback are in the usage of the callback, which should be updated in #1401.

**Review by chynesNR — APPROVED** [review]

**Review by tippmar-nr — APPROVED** [review]

**Comment by nrcventura** [comment]
The CSP/HSM test failures are not related to the changes in this PR, and are being handled separately.

**Pull request merged**